src/domain/human-description/entities/human-description-template.entity.ts

```diff
@@ -5,9 +5,17 @@ import {
   HumanDescriptionFragment,
   TokenValueFragment,
   ValueType,
-  WordFragment,
+  TextFragment,
 } from './human-description.entity';
 
+type SafeRegExpMatchArray = RegExpMatchArray & {
+  groups: NonNullable<RegExpMatchArray['groups']>;
+};
+
+function hasGroups(match: RegExpMatchArray): match is SafeRegExpMatchArray {
+  return !!match.groups;
+}
```

Comment:
I don't think this is clear. Do we want to check if it's not undefined? How about:

```typescript
return match.groups !== undefined
```